### PR TITLE
RSPEED-2928: Add optional Sentry error tracking integration

### PR DESCRIPTION
**Pull request by major**
## Description

Add optional Sentry error tracking to lightspeed-stack. When a `SENTRY_DSN` environment variable is set, the service initializes the Sentry SDK at startup to capture unhandled exceptions and sample POST request performance traces. When the variable is unset, Sentry is completely disabled and the service runs normally.

Key behaviors:
- Health check, metrics, and root routes are excluded from trace sampling
- Only POST requests are captured via the FastAPI integration
- `send_default_pii` is disabled
- Optional `SENTRY_CA_CERTS` support for private/internal CA Sentry instances
- Init failures are caught and logged without crashing the service
- Pending events are flushed on shutdown after cleanup tasks complete
- Default environment is `development` so production deployers must set `SENTRY_ENVIRONMENT` explicitly

## Type of change

- [x] New feature
- [x] Documentation Update

## Tools used to create PR

- Assisted-by: Claude (OpenCode/Sisyphus), CodeRabbit
- Generated by: N/A

## Related Tickets & Documents

- [RSPEED-2928](https://redhat.atlassian.net/browse/RSPEED-2928)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [x] If it is a core feature, I have added thorough tests.

## Testing

1. Run the Sentry unit tests:
   ```bash
   uv run pytest tests/unit/test_sentry.py -v
   ```
   All 18 tests pass, covering: DSN set/unset, empty DSN, CA cert exists/missing, custom/default environment, init failure handling, trace sampler for excluded/normal routes, and edge cases (empty context, missing path, None path).

2. To verify runtime behavior, set `SENTRY_DSN` to a valid DSN and start the service. Check logs for `Sentry initialized`. Unset the variable and confirm `Sentry DSN not configured, skipping initialization`.

## Release Notes

* **New Features**
  * Sentry error tracking is now available as an optional feature, with configurable sampling rates and excluded routes.

* **Documentation**
  * Sentry integration guide added with configuration instructions for environment variables and OpenShift deployment.

* **Chores**
  * Added Sentry SDK dependency.

* **Tests**
  * Added test coverage for Sentry initialization and trace sampling behavior.

### CODE REVIEW

File: src/app/main.py [hunk 1]
```diff
@@ -83,6 +86,12 @@ async def lifespan(_app: FastAPI) -> AsyncIterator[None]:
     # Cleanup resources on shutdown
     await shutdown_background_topic_summary_tasks()
     await A2AStorageFactory.cleanup()
+
+    # Flush pending Sentry events after cleanup so any errors during
+    # shutdown are captured before the process exits.
+    import sentry_sdk  # pylint: disable=import-outside-toplevel
+
+    sentry_sdk.flush(timeout=2)
     logger.info("App shutdown complete")
```
[the remark this answers is not in the document]
Comment (major) [reply src/app/main.py:95]: Fixed. Wrapped cleanup in `try`/`finally` so `sentry_sdk.flush()` always runs even if cleanup raises. Also moved the `sentry_sdk` import to top-level (see next comment).

File: src/app/main.py [hunk 2]
```diff
@@ -83,6 +86,12 @@ async def lifespan(_app: FastAPI) -> AsyncIterator[None]:
     # Cleanup resources on shutdown
     await shutdown_background_topic_summary_tasks()
     await A2AStorageFactory.cleanup()
+
+    # Flush pending Sentry events after cleanup so any errors during
+    # shutdown are captured before the process exits.
+    import sentry_sdk  # pylint: disable=import-outside-toplevel
+
+    sentry_sdk.flush(timeout=2)
```
[the remark this answers is not in the document]
Comment (major) [reply src/app/main.py]: Fixed. Moved `import sentry_sdk` to top-level module imports with the same `pyright: ignore[reportMissingImports]` as `sentry.py`. Removed the inline import and `pylint: disable=import-outside-toplevel`.

File: src/sentry.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    """
+    asgi_scope = tracing_context.get("asgi_scope", {})
+    path = asgi_scope.get("path") if isinstance(asgi_scope, dict) else None
+
+    if path is not None and any(
+        path.endswith(route) for route in SENTRY_EXCLUDED_ROUTES
+    ):
+        return 0.0
+
+    return SENTRY_DEFAULT_TRACES_SAMPLE_RATE
```
[the remark this answers is not in the document]
Comment (major) [reply src/sentry.py]: Good catch. Fixed - root path `"/"` is now exact-matched separately, and `endswith` only applies to non-root routes. Added a trailing-slash test case (`"/v1/query/"`) confirming it returns the default sample rate.

File: tests/unit/test_sentry.py [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+    def test_missing_path_in_asgi_scope(self) -> None:
+        """Test that missing path key in asgi_scope returns the default rate."""
+        context: dict = {"asgi_scope": {}}
+        assert sentry_traces_sampler(context) == SENTRY_DEFAULT_TRACES_SAMPLE_RATE
+
+    def test_none_path_value(self) -> None:
+        """Test that a None path value returns the default sample rate."""
+        context: dict = {"asgi_scope": {"path": None}}
+        assert sentry_traces_sampler(context) == SENTRY_DEFAULT_TRACES_SAMPLE_RATE
```
[the remark this answers is not in the document]
Comment (major) [reply tests/unit/test_sentry.py:186]: Fixed alongside the sampler logic. Added `"/v1/query/"` to the parametrized `test_normal_routes_return_default_rate` test - 19/19 tests pass.

**Review by tisnik — APPROVED** [review]
LGTM
